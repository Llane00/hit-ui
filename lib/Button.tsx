import React, { useState } from "react";

function Button() {
  const [n, setN] = useState(0);

  function handleClick() {
    console.log(1)
    setN(n + 1)
  }

  return (
    <div>
      <div>number: {n}</div>
      <button onClick={handleClick}>add 1</button>
    </div>
  );
}

export default Button;