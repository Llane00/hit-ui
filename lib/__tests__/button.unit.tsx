import Button from "../Button"
import React from "react"
import renderer from "react-test-renderer"

describe('button', () => {
  it('是一个div', () => {
    const tree = renderer
        .create(<Button/>)
        .toJSON()
    expect(tree).toMatchSnapshot()
  })
})