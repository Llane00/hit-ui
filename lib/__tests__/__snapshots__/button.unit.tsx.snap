// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 是一个div 1`] = `
<div>
  <div>
    number: 
    0
  </div>
  <button
    onClick={[Function]}
  >
    add 1
  </button>
</div>
`;
